src/crypto/tls/handshake_server_tls13.go

```diff
@@ -895,6 +947,8 @@ func (hs *serverHandshakeStateTLS13) readClientFinished() error {
 		return unexpectedMessageError(finished, msg)
 	}
 
+	hs.handshakeTimings.ReadClientFinished = hs.handshakeTimings.elapsedTime()
```

Comment:
This should come after verifying the MAC (i.e., at the end of this function).

Reply:
This one I disagree with. It shouldn't go *quite* at the end.